### ISSUE TITLE
Clicking the text at a checkbox does not toggle the checkbox

### ISSUE THREAD

**Issue by SimpelMe** [hedgedoc/hedgedoc#6828]
### Description

When clicking the text at a checkbox you won't toggle the checkbox itself. This is an a11y issue. The text at the checkbox should be a label for the checkbox. Current code looks like this:
```html
<li class="task-list-item" data-startline="3" data-endline="3">
    <input type="checkbox" class="task-list-item-checkbox">
    <label></label>Item 1
</li>
```
There is a label but it's empty and preceeding the text. Second the label must wrap the input or referencing the checkbox by it's id (not existing yet).

### Steps to reproduce

1. create a checkbox including text
2. click the text

### Expected behaviour

Clicking the text ticks the checkbox.

### Logs

```shell

```

### Config

```shell

```

### Your Setup

- tested with current https://demo.hedgedoc.org
- HedgeDoc version: 1.12.0
